### PR TITLE
feat: show progress when creating Lib zip file

### PR DESCRIPTION
**Pull request by gerard-queralt**
As requested, we now show the percentage of files that have been zipped when building the game. Should be easy enought to also estimate the remaining time, if someone wants to do it 👀

### CODE REVIEW

**Comment by gerard-queralt** [comment]
Might as well have named this to "Why we need #502 yesterday" 😬

File: Source/Auxiliar/GameBuilder.cpp [hunk 1]
```diff
@@ -77,18 +84,44 @@ void AddConfigToZip(const std::string& startingScene)
...
+{
+	LOG_VERBOSE("Item {} zipped in {} seconds ({}/{})",
+				data.fileZipped,
+				data.timeTaken.count(),
+				data.fileZippedIndex,
+				data.totalFiles);
+	std::chrono::minutes minutesToZip = std::chrono::duration_cast<std::chrono::minutes>(data.timeTaken);
+	if (minutesToZip >= std::chrono::minutes(maxMinutesToZip))
+	{
+		LOG_WARNING("Item {} took more that {} minute(s) to zip. Seconds it took: {}",
```
Comment (MarcGonzalezMoratona) [inline Source/Auxiliar/GameBuilder.cpp:101]:
```suggestion
		LOG_WARNING("Item {} took more than {} minute(s) to zip. Seconds it took: {}",
```

**Comment by MarcGonzalezMoratona** [comment]
I want to share some feedback comments about the user experience of this feature. I see that the window has a resize option on the bottom-right corner, but it looks like it does not work. I see that the window is a bit small and I can't see the whole text

File: Source/DataModels/Windows/PopUpWindows/WindowLoading.cpp [hunk 2]
```diff
@@ -5,7 +5,7 @@
 
 WindowLoading::WindowLoading() : PopUpWindow("Loading")
```
Comment (MarcGonzalezMoratona) [inline Source/DataModels/Windows/PopUpWindows/WindowLoading.cpp:6]:
```suggestion
WindowLoading::WindowLoading() : PopUpWindow("Loading...")
```

File: Source/Auxiliar/GameBuilder.cpp [hunk 3]
```diff
@@ -77,18 +84,44 @@ void AddConfigToZip(const std::string& startingScene)
...
 	std::string path = GAME_STARTING_CONFIG;
 
-	App->GetModule<ModuleFileSystem>()->AppendToZipFolder(zipPath, path.c_str(), buffer.GetString(), buffer.GetSize(), true);
+	App->GetModule<ModuleFileSystem>()->AppendToZipFolder(
+		zipPath, path.c_str(), buffer.GetString(), buffer.GetSize(), true);
+}
+
+void OnFileZipped(const FileZippedData& data)
+{
+	LOG_VERBOSE("Item {} zipped in {} seconds ({}/{})",
```
Comment (MarcGonzalezMoratona) [inline Source/Auxiliar/GameBuilder.cpp:93]:
```suggestion
	LOG_VERBOSE("File {} zipped in {} seconds ({}/{})",
```

**Review by MarcGonzalezMoratona — DISMISSED** [review]
This feature looks amazing. Thank you!

**Comment by gerard-queralt** [comment]
> I see that the process can't be cancelled, once you are building you have to wait or close the engine. For this reason, maybe the cross on the corner of the window makes no sense, but it's not a big issue

I'm not sure hiding the close "X" is possible, and haven't managed to find anything that suggests the opposite. Also, I'd like to make the zip operation cancellable at some point, but that's quite tricky to do so it's likely we won't have it in the end.

**Review by MarcGonzalezMoratona — DISMISSED** [review]

**Review by Dedado3000 — APPROVED** [review]

**Review by MarcGonzalezMoratona — APPROVED** [review]

**Pull request merged**